packages/nimble-components/src/rich-text/mention-list-box/index.ts

```diff
@@ -0,0 +1,294 @@
+        this.filterOptions();
+    }
+
+    public onMention(props: SuggestionProps): void {
+        this.suggestionProps = props;
+        this.setActiveCharacter(props.text.slice(0, 1));
+        this.filter = props.query;
+        this.setAnchorElement(props.decorationNode as HTMLElement);
+        this.setOpen(true);
+        void this.selectFirstOptionIfValidOptionExists();
+    }
+
+    public close(): void {
+        this.setOpen(false);
+    }
+
+    /**
+     * @internal
+     */
+    public activeCharacterChanged(): void {
+        this.setActiveMappingConfigs();
+    }
+
+    /**
+     * @internal
+     */
+    public filterChanged(): void {
+        this.filterOptions();
+    }
+
+    /**
+     * @internal
+     */
+    public childItemsChanged(): void {
+        void this.selectFirstOptionIfValidOptionExists();
+    }
+
+    /**
+     * @internal
+     */
+    public anchorElementChanged(prev: HTMLElement, next: HTMLElement): void {2
+        if (prev) {
+            this.intersectionObserver.unobserve(prev);
+        }
+        if (this.region && this.anchorElement) {
+            this.region.anchorElement = this.anchorElement;
+            this.region.update();
+            this.intersectionObserver.observe(next);
+        }
+    }
+
+    private setActiveCharacter(activeCharacter: string): void {
+        this.activeCharacter = activeCharacter;
+    }
+
+    private setAnchorElement(anchorElement: HTMLElement): void {
+        this.anchorElement = anchorElement;
+    }
+
+    private scrollOptionIntoView(): void {
+        this.listBox?.firstSelectedOption?.scrollIntoView({ block: 'nearest' });
+    }
+
+    private setActiveMappingConfigs(): void {
+        this.activeMappingConfigs = this.activeCharacter ? this.mentionExtensionConfig?.find(config => config.character === this.activeCharacter)?.mappingConfigs : undefined;
+    }
+
+    private filterOptions(): void {
+        if (!this.childItems || this.filter === undefined) {
+            return;
+        }
+        const normalizedFilter = normalizeString(this.filter);
+        this.childItems.forEach(element => {
+            const listOption = element as ListOption;
+            const normalizedText = normalizeString(listOption.text);
+            const checkFlag = !normalizedText.includes(normalizedFilter);
+            listOption.disabled = checkFlag;
+            listOption.hidden = checkFlag;
+        });
+        void this.selectFirstOptionIfValidOptionExists();
+    }
+
+    private activateMention(mentionDetail: MentionDetail): void {
+        this.suggestionProps.command({
+            href: mentionDetail.href,
+            label: mentionDetail.displayName
+        });
+        this.setOpen(false);
+    }
+
+    private setOpen(value: boolean): void {
+        this.open = value;
+    }
+
+    private updateSelectableOption(): void {
+        this.hasAnySelectableOption = !this.childItems.map(element => element as ListOption).every(
```

Comment:
It seems like you might not need this. The implementation of `selectFirstOption` already [excludes disabled options](https://github.com/microsoft/fast/blob/fd65254531ac3a593b32528d81b44ebf82c8e4ab/packages/web-components/fast-foundation/src/listbox/listbox.ts#L483), and `firstSelectedOption` [returns `null` if there is no selected option.](https://github.com/microsoft/fast/blob/fd65254531ac3a593b32528d81b44ebf82c8e4ab/packages/web-components/fast-foundation/src/listbox/listbox.ts#L40) Because of that, `scrollOptionIntoView` is safe to call even if there are no selectable options. If you can do without `hasAnySelectableOption`, that would be nice.

Unfortunately, I just noticed that `firstSelectedOption` is documented as internal. To be safe, I would probably just copy the implementation, since it is so simple:
```TS
    public get firstSelectedOption(): ListOption {
        return this.selectedOptions[0] ?? null;
    }
```

Reply:
Updated as per the suggestion